aeolis/tests/test_utils.py

```diff
@@ -0,0 +1,44 @@
+"""
+This file is part of AeoLiS test suit. It uses the Pytest framework.
+
+To run all tests, use:
+    pytest
+
+To run only the test in this file, use:
+    pytest aeolis/tests/test_utils.py
+
+To run all test in a group (class), use:
+    pytest aeolis/tests/test_utils.py::<TestClass>
+
+To run an specific test, use:
+    - If in a group:
+        pytest aeolis/tests/test_utils.py::<TestClass>::<test-method>
+    
+    - If not in any group:
+        pytest aeolis/tests/test_utils.py::<test-function>
+"""
+
+from aeolis import utils
+import numpy as np
+
+
+# Use test classes to group test-cases that belong to the 
+# same scenario; unless a single test-case is required
+
+class TestIsIterable:
+    """
+    Test if iterable variables can be correctly distinguished from non-iterable variao bles
+    """
+    def test_isiterable(self):
+        x_iterable = np.random.rand(3,2)
+        """
+        Check if an inerable variable returns True.
+        """
+        assert utils.isiterable(x_iterable) == True
+
+    def test_not_iterable(self):
+        """Check if non-iterable variable returns False"""
+        x_iterable = None
+        assert utils.isiterable(x_iterable) == False
```

Comment:
The negative test case can be included in the `test_isiterable()` function and the function `test_not_isiterable` can be removed. The class `TestIsIterable` then becomes obsolete and can be removed; the function `test_isiterable` can exists as a standalone function outside any class.

Reply:
That's right. The purpose of this code is an example of how test-scenarios can be organized as classes and the test cases as methods in that class. However, trivial, I would like to keep this code a generic example.

Reply:
As we discussed @manuGil , the two tests can be combined to one by including the negative and positive tests cases in one function, but the purpose of this example is to demonstrate the grouping of related tests in a class. The idea of grouping related tests in a class would help break down the test for a large function into several shorter tests.

Reply:
Thanks for the explanation.
The idea of grouping related tests in a class would help in writing tests for a large function as several short tests, so it would be good to demonstrate that in an example such as this.